test/EFCore.PG.FunctionalTests/BulkUpdates/ComplexTypeBulkUpdatesNpgsqlTest.cs

```diff
@@ -109,7 +109,8 @@ public override async Task Update_complex_type_to_parameter(bool async)
             """
 @complex_type_p_AddressLine1='New AddressLine1'
 @complex_type_p_AddressLine2='New AddressLine2'
-@complex_type_p_Tags={ 'new_tag1', 'new_tag2' } (DbType = Object)
+@complex_type_p_Tags={ 'new_tag1'
```

Comment:
@cincuranet see this change after syncing to latest upstream EF; this is most likely because of https://github.com/dotnet/efcore/pull/36594. I think we should improve the logic slightly.

Reply:
Well, couldn't we e.g. track parentheses characters in the logic you added ('{', '(', '[') and ignore commas when they're inside?